### PR TITLE
[build][c10] set up c10 scaffolding. Move macros proper first.

### PR DESCRIPTION
**Pull request by Yangqing**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,12 @@ cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
 # ---[ Project and semantic versioning.
 project(Caffe2 CXX C)
 
+set(CMAKE_CXX_STANDARD 11)
+if (NOT MSVC)
+  set(CMAKE_C_STANDARD 11)
+endif()
+set(CMAKE_VERBOSE_MAKEFILE ON)
```
Comment (Yangqing) [inline CMakeLists.txt]: (note: this will be removed when the pr is ready to be merged)

File: aten/src/ATen/core/Macros.h [hunk 2]
```diff
@@ -3,41 +3,7 @@
...
-// AT_CORE_IMPORT.
-//
-
-// TODO: unify the controlling macros.
-#if defined(CAFFE2_BUILD_MAIN_LIBS) || defined(ATen_cpu_EXPORTS) || defined(caffe2_EXPORTS)
-#define AT_CORE_API AT_CORE_EXPORT
-#else // defined(CAFFE2_BUILD_MAIN_LIBS) || defined(ATen_cpu_EXPORTS) || defined(caffe2_EXPORTS)
-#define AT_CORE_API AT_CORE_IMPORT
-#endif // defined(CAFFE2_BUILD_MAIN_LIBS) || defined(ATen_cpu_EXPORTS) || defined(caffe2_EXPORTS)
+#include "c10/macros/Macros.h"
```
Comment (ezyang) [inline aten/src/ATen/core/Macros.h]: What's the rule for subfolders under c10? My preference (evidenced by `ATen/core/` structure) is that most headers live in the top level.

Comment (Yangqing) [reply aten/src/ATen/core/Macros.h]: At some point we should just include a unified c10/c10.h that includes all dependent headers, and have c10 manage the internal header relationships - does that make sense? I think eigen uses this as well by including headers like <Eigen/Core> without exposing further details.

**Comment by Yangqing** [comment]
OK, this one should be correct because last run had mac errors due to ci system, this time mac passed, and mkl is erroring out due to timeout, but mkl already passed last time. No code change between the two runs.

**Pull request closed without merging**